inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -125,10 +125,44 @@ class CNNLayerCreator : public ::ngraph::AttributeVisitor {
 
     void on_adapter(const std::string& name, ::ngraph::ValueAccessor<void>& adapter) override;
 
+    void on_adapter(const std::string& name, ::ngraph::ValueAccessor<void*>& adapter) override {
+        auto data = reinterpret_cast<char*>(adapter.get_ptr());
+        std::vector<char> dataBuff(data, data + adapter.size());
+        std::stringstream ss;
+        copy(dataBuff.begin(), dataBuff.end(), std::ostream_iterator<char>(ss, ","));
```

Comment:
When NodeConverter is removed for Const op, "on_adapter" is required for converting Function to CNN network as it calls node->visit_attributes(visitor) to acquire constant data.

Reply:
Where do you call this adapter? I am not sure that it works as expected.

I think you use `shareWeights` when you convert constant to CNNLayer.

Is it needed only in order to [visit_attribute for Constant](https://github.com/openvinotoolkit/openvino/pull/2573/files#diff-742b80519ee60a39f712a40150b2840f4629c5894aaa1456186feda1e41c36ddR796) returns true?

Reply:
It is called for CNNLayerCreator visitor:
            CNNLayerCreator visitor(node);
            if (node->visit_attributes(visitor)) result = visitor.create();
Purpose of this method is to convert Const data to string with "," separation.  CNNLayerCreator has member variable "param" which stores all op attributtes in string format.
shareWeights function is used in "addSpecificCreator" when CNNLayerCreator creates Const op.

Reply:
But we don't need to do it, because:

- CNNLayer Const layer doesn't have data in the parameter (for it we use blobs map).
- For some custom operations which (as we can understand from the on_adapter for reader) has a bytes array in the attribute, it also incorrect because you need to save byte array to parameter map (but you save an array in human readable format).

Reply:
Corrected, on_adpater(void*) does not copy bytes array for Const op, it will throw an exception if op (!=Const) will try to use this method. According to our agreement new task is created in order to implement 'on_adapter(void*)' and add test for this use case.